### ISSUE TITLE
[Bug]: Unicode handling in VS Code on windows 11

### ISSUE THREAD

**Issue by someaka** [Pythagora-io/gpt-pilot#466]
### Version

VisualStudio Code extension

### Operating System

Windows 11

### What happened?

GPT Pilot was chugging along nicely when it decided to use a unicode character, the rocket  🚀

`..... Good luck with the development of "QuickBlur"! 🚀
Saving file c:\Users\M\Desktop\pilot\gpt-pilot\workspace\QuickBlur.env

Saving file c:\Users\M\Desktop\pilot\gpt-pilot\workspace\QuickBlur\index.js

Saving file c:\Users\M\Desktop\pilot\gpt-pilot\workspace\QuickBlur\package.json

Saving file c:\Users\M\Desktop\pilot\gpt-pilot\workspace\QuickBlur\tests\testAuth.js

Saving file c:\Users\M\Desktop\pilot\gpt-pilot\workspace\QuickBlur\views\login.ejs

Dev step 113

---------- GPT PILOT EXITING WITH ERROR ----------

Traceback (most recent call last):

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\main.py", line 78, in <module> project.start()

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\Project.py", line 141, in start self.developer.start_coding()

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\agents\Developer.py", line 70, in start_coding self.implement_task(i, dev_task)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\agents\Developer.py", line 117, in implement_task result = self.execute_task(convo_dev_task, ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\agents\Developer.py", line 455, in execute_task return self.task_postprocessing(convo, development_task, continue_development, result, function_uuid) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\agents\Developer.py", line 317, in task_postprocessing return self.continue_development(convo, last_branch_name, continue_description, development_task) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\agents\Developer.py", line 497, in continue_development iteration_description = iteration_convo.send_message('development/iteration.prompt', { ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\AgentConvo.py", line 111, in send_message self.log_message(message_content)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\helpers\AgentConvo.py", line 246, in log_message print(f"\n{content}\n", type='local')

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\utils\custom_print.py", line 16, in print_to_external_process local_print(*args, **kwargs)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot\utils\custom_print.py", line 33, in local_print built_in_print(message, **kwargs)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot-env\Lib\site-packages\colorama\ansitowin32.py", line 47, in write self.__convertor.write(text)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot-env\Lib\site-packages\colorama\ansitowin32.py", line 177, in write self.write_and_convert(text)

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot-env\Lib\site-packages\colorama\ansitowin32.py", line 205, in write_and_convert self.write_plain_text(text, cursor, len(text))

File "c:\Users\M\Desktop\pilot\gpt-pilot\pilot-env\Lib\site-packages\colorama\ansitowin32.py", line 210, in write_plain_text self.wrapped.write(text[start:end])

File "C:\Users\M\AppData\Local\Programs\Python\Python312\Lib\encodings\cp1252.py", line 19, in encode return codecs.charmap_encode(input,self.errors,encoding_table)[0] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f680' in position 3906: character maps to \<undefined\>

--------------------------------------------------`

TLDR the  🚀 makes the VS extension crash

tried entering 
'chcp 65001'
 in the powershell before running the extension but to no avail

**Comment by someaka** [comment]
quick work around :
modify the custom_print.py

    def remove_unicode_encode(text):
        return text.encode('ascii', 'ignore').decode('ascii')

 

    def local_print(*args, **kwargs):
        message = remove_unicode_encode( " ".join(map(str, args)) )
        if 'type' in kwargs:
            if kwargs['type'] in LOCAL_IGNORE_MESSAGE_TYPES:
                return
            del kwargs['type']

        built_in_print(message, **kwargs)

**Comment by senko** [comment]
Marking as duplicate of #509
